File: PrivacifyDoc/PrivacifyDocApi/Domain/Models/Document.cs
using System;
using System.Collections.Generic;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace PrivacifyDoc.Domain.Models
{
    public class Document
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; } = null!;
        
        public string OriginalFileName { get; set; } = null!;
        
        public string ContentType { get; set; } = null!;
        
        public DateTime UploadDate { get; set; }
        
        public ProcessingStatus Status { get; set; }
        
        public string? ErrorMessage { get; set; }

        public string OriginalFilePath { get; set; } = null!;
        
        public string? AnonymizedFilePath { get; set; }
        
        public AnonymizationSettings Settings { get; set; } = new();
        
        public List<SensitiveData> DetectedSensitiveData { get; set; } = new();
    }

    public enum ProcessingStatus
    {
        Uploaded,
        Processing,
        OCRCompleted,
        DataDetected,
        Anonymized,
        Error
    }

    public class AnonymizationSettings
    {
        public List<SensitiveDataType> DataTypesToAnonymize { get; set; } = new();
        
        public AnonymizationMethod Method { get; set; }
        
        public Dictionary<string, string> Options { get; set; } = new();
    }

    public enum AnonymizationMethod
    {
        Masking,
        Redaction,
        Pseudonymization,
        Removal
    }

    public enum SensitiveDataType
    {
        PersonName,
        Address,
        PhoneNumber,
        Email,
        PESEL,
        NIP,
        REGON,
        IDNumber,
        PassportNumber,
        BankAccount,
        CreditCard,
        MedicalData,
        Custom
    }

    public class SensitiveData
    {
        public SensitiveDataType Type { get; set; }
        
        public string OriginalValue { get; set; } = null!;
        
        public string AnonymizedValue { get; set; } = null!;
        
        public DataLocation Location { get; set; } = null!;
        
        public double Confidence { get; set; }
    }

    public class DataLocation
    {
        public int? PageNumber { get; set; }
        
        public Rectangle? BoundingBox { get; set; }
        
        public int? StartIndex { get; set; }
        
        public int? EndIndex { get; set; }
    }

    public class Rectangle
    {
        public int X { get; set; }
        public int Y { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }
    }
    
    public class DocumentAnonymizationStats
    {
        public string DocumentId { get; set; } = null!;
        public int DetectedDataCount { get; set; }
    }
}

File: frontend/src/types/document.ts
export interface Document {
  id: string;
  originalFileName: string;
  contentType: string;
  uploadDate: string;
  status: ProcessingStatus;
  errorMessage?: string;
  originalFilePath: string;
  anonymizedFilePath?: string;
  detectedSensitiveData: SensitiveData[];
}

export const ProcessingStatus = {
  Uploaded: 'Uploaded',
  Processing: 'Processing',
  OCRCompleted: 'OCRCompleted',
  DataDetected: 'DataDetected',
  Anonymized: 'Anonymized',
  Error: 'Error'
} as const;

export type ProcessingStatus = typeof ProcessingStatus[keyof typeof ProcessingStatus];

export interface SensitiveData {
  type: SensitiveDataType;
  originalValue: string;
  anonymizedValue: string;
  confidence: number;
}

export const SensitiveDataType = {
  PersonName: 'PersonName',
  Address: 'Address',
  PhoneNumber: 'PhoneNumber',
  Email: 'Email',
  PESEL: 'PESEL',
  NIP: 'NIP',
  REGON: 'REGON',
  IDNumber: 'IDNumber',
  PassportNumber: 'PassportNumber',
  BankAccount: 'BankAccount',
  CreditCard: 'CreditCard',
  MedicalData: 'MedicalData',
  Custom: 'Custom'
} as const;

export type SensitiveDataType = typeof SensitiveDataType[keyof typeof SensitiveDataType];

export interface DocumentStatusResponse {
  documentId: string;
  status: ProcessingStatus;
  progress: number;
  detectedDataCount: number;
  errorMessage?: string;
}

export interface UploadResponse {
  documentId: string;
}